### PR TITLE
[22.05] netatalk: FreeBSD patchset 3.1.13_3, critical fix

### PR DESCRIPTION
**Pull request by raindev**
###### Description of changes

A backport of #196933 and #199056 with a fix for a segmentation fault in afpd.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This should include #199056 if it gets merged

**Comment by raindev** [comment]
Is there anything else to be done to be able to merge the PR?

File: pkgs/tools/filesystems/netatalk/default.nix [hunk 1]
```diff
@@ -1,7 +1,6 @@
-{ fetchurl, lib, stdenv, autoreconfHook, pkg-config, perl, python3
-, db, libgcrypt, avahi, libiconv, pam, openssl, acl
-, ed, libtirpc, libevent, fetchpatch
-}:
+{ fetchurl, lib, stdenv, autoreconfHook, pkg-config, perl, python3, db
```
Comment (AndersonTorres) [inline pkgs/tools/filesystems/netatalk/default.nix:1]: One per line, asciibetical ordering, starting with 

```
{ lib
, stdenv
, fetchurl
, fetchpatch
, ...
}
```

File: pkgs/tools/filesystems/netatalk/default.nix [hunk 2]
```diff
@@ -22,6 +21,69 @@ stdenv.mkDerivation rec {
...
+      sha256 = "sha256-lFWF0Qo8PJv7QKvnMn0Fc9Ruzb+FTEWgOMpxc789jWs=";
+    })
+  ];
+
+  postPatch = ''
+    # freeBSD patches are -p0
+    for i in $freeBSDPatches ; do
+      patch -p0 < $i
+    done
+  '';
```
Comment (AndersonTorres) [inline pkgs/tools/filesystems/netatalk/default.nix:85]:
```suggestion
  patchFlags = [ "-p0" ];
  '';
```

File: pkgs/tools/filesystems/netatalk/default.nix [hunk 3]
```diff
@@ -22,6 +21,69 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  freeBSDPatches = [
+    # https://bugs.freebsd.org/263123
+    (fetchpatch {
+      name = "patch-etc_afpd_directory.c";
```
Comment (AndersonTorres) [inline pkgs/tools/filesystems/netatalk/default.nix:27]: Remove the `name` s, they are not needed.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by raindev** [comment]
@AndersonTorres thanks for the review! I'm happy to do the changes, but shouldn't they be applied in `master` rather than as a part of the backport?

**Comment by AndersonTorres** [comment]
Is this a backport??

Oh, forget it then

**Pull request merged**